controllers/ocirepository_controller_test.go

```diff
@@ -240,265 +243,374 @@ func TestOCIRepository_Reconcile(t *testing.T) {
+			},
 		},
 		{
-			name:                  "private-registry-access-via-serviceaccount",
-			url:                   ociURL,
-			digest:                podinfoImageDigest,
-			includeSecretRef:      false,
-			includeServiceAccount: true,
+			name:    "HTTPS with invalid certfile",
+			want:    sreconcile.ResultEmpty,
+			wantErr: true,
+			registryOpts: registryOptions{
+				withTLS: true,
+			},
+			craneOpts: []crane.Option{crane.WithTransport(&http.Transport{
+				TLSClientConfig: &tls.Config{
+					RootCAs: pool,
+				},
+			}),
+			},
+			tlsCertSecret: &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "ca-file",
+				},
+				Data: map[string][]byte{
+					"caFile": []byte("invalid"),
+				},
+			},
+			assertConditions: []metav1.Condition{
+				*conditions.TrueCondition(sourcev1.FetchFailedCondition, sourcev1.OCIOperationFailedReason, "failed to pull artifact from "),
+			},
 		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			g := NewWithT(t)
 
-			ns, err := testEnv.CreateNamespace(ctx, "ocirepository-test")
-			g.Expect(err).ToNot(HaveOccurred())
-			defer func() { g.Expect(testEnv.Delete(ctx, ns)).To(Succeed()) }()
-
-			secret := &corev1.Secret{
-				ObjectMeta: metav1.ObjectMeta{
-					GenerateName: "auth-secretref",
-					Namespace:    ns.Name,
-				},
-				Type: corev1.SecretTypeDockerConfigJson,
-				StringData: map[string]string{
-					".dockerconfigjson": fmt.Sprintf(`{"auths": {%q: {"username": %q, "password": %q}}}`, repositoryURL, testRegistryUsername, testRegistryPassword),
-				},
-			}
-			g.Expect(testEnv.CreateAndWait(ctx, secret)).To(Succeed())
-			defer func() { g.Expect(testEnv.Delete(ctx, secret)).To(Succeed()) }()
-
-			serviceAccount := &corev1.ServiceAccount{
-				ObjectMeta: metav1.ObjectMeta{
-					GenerateName: "sa-ocitest",
-					Namespace:    ns.Name,
-				},
-				ImagePullSecrets: []corev1.LocalObjectReference{{Name: secret.Name}},
-			}
-			g.Expect(testEnv.CreateAndWait(ctx, serviceAccount)).To(Succeed())
-			defer func() { g.Expect(testEnv.Delete(ctx, serviceAccount)).To(Succeed()) }()
+			builder := fakeclient.NewClientBuilder().WithScheme(testEnv.GetScheme())
 
 			obj := &sourcev1.OCIRepository{
 				ObjectMeta: metav1.ObjectMeta{
-					GenerateName: "ocirepository-test-resource",
-					Namespace:    ns.Name,
+					GenerateName: "auth-strategy-",
 				},
 				Spec: sourcev1.OCIRepositorySpec{
-					URL:       tt.url,
-					Interval:  metav1.Duration{Duration: 60 * time.Minute},
-					Reference: &sourcev1.OCIRepositoryRef{Digest: tt.digest.String()},
+					Interval: metav1.Duration{Duration: interval},
+					Timeout:  &metav1.Duration{Duration: timeout},
 				},
 			}
 
-			if tt.includeSecretRef {
-				obj.Spec.SecretRef = &meta.LocalObjectReference{Name: secret.Name}
-			}
+			server, err := setupRegistryServer(ctx, tt.registryOpts)
```

Comment:
It may be better to add a new workdir field in the `registryOptions`. If it's set, it's used, else a tempdir is created.
In TestMain(), since we don't have `testing.T`, we can let the tempdir be created and deleted at the end of TestMain().
But for the tests, we can create a tempdir using `t.TempDir()` and set this value in `registryOption`. At the end of the test, the directory would be automatically deleted.

Reply:
> . If it's set, it's used, else a tempdir is created

I think that we can leave the creation and deletion of the workspace directory up to the caller of the function. We can take it as an arg and return an error if it is empty.
Wdyt?

Reply:
That'd be okay. The previous recommendation is based on not trying to change much of the existing behavior, but sounds okay to me.